From 18178ba09b221eef7f0bb869980e1c043a8e764f Mon Sep 17 00:00:00 2001
From: Rob Crittenden <rcritten@redhat.com>
Date: Wed, 31 May 2023 17:21:55 -0400
Subject: [PATCH] Address issues uncovered by pylint 2.15.5

Two variables used before assignment

Three Useless suppression of 'unexpected-keyword-arg'

Fixes: https://github.com/freeipa/freeipa-healthcheck/issues/295

Signed-off-by: Rob Crittenden <rcritten@redhat.com>
---
 src/ipahealthcheck/ipa/certs.py | 5 +----
 src/ipahealthcheck/ipa/trust.py | 2 +-
 2 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/src/ipahealthcheck/ipa/certs.py b/src/ipahealthcheck/ipa/certs.py
index 11ac0c1..4ea5112 100644
--- a/src/ipahealthcheck/ipa/certs.py
+++ b/src/ipahealthcheck/ipa/certs.py
@@ -343,7 +343,6 @@ class IPACertfileExpirationCheck(IPAPlugin):
 
                 try:
                     if 'pwd_file' in signature(certdb.NSSDatabase).parameters:
-                        # pylint: disable=unexpected-keyword-arg
                         db = certdb.NSSDatabase(
                             dbdir, token=token,
                             pwd_file=pwd_file.name if pwd_file else None)
@@ -624,7 +623,6 @@ class IPACertNSSTrust(IPAPlugin):
             pwd_file = get_token_password_file(self.ca.hsm_enabled,
                                                token)
 
-            # pylint: disable=unexpected-keyword-arg
             db = certdb.NSSDatabase(
                 paths.PKI_TOMCAT_ALIAS_DIR, token=token,
                 pwd_file=pwd_file.name if pwd_file else None)
@@ -987,7 +985,7 @@ class IPANSSChainValidation(IPAPlugin):
                         key=key,
                         dbdir=dbdir,
                         nickname=nickname,
-                        reason=response.output_error,
+                        reason=str(e),
                         msg='Validation of {nickname} in {dbdir} failed: '
                             '{reason}')
                 else:
@@ -1251,7 +1249,6 @@ class IPACertRevocation(IPAPlugin):
                 dbdir = request.get('cert-database')
                 try:
                     if 'pwd_file' in signature(certdb.NSSDatabase).parameters:
-                        # pylint: disable=unexpected-keyword-arg
                         db = certdb.NSSDatabase(
                             dbdir, token=token,
                             pwd_file=pwd_file.name if pwd_file else None
diff --git a/src/ipahealthcheck/ipa/trust.py b/src/ipahealthcheck/ipa/trust.py
index b962807..243502f 100644
--- a/src/ipahealthcheck/ipa/trust.py
+++ b/src/ipahealthcheck/ipa/trust.py
@@ -307,7 +307,7 @@ class IPATrustCatalogCheck(IPAPlugin):
                 id = pysss_nss_idmap.getnamebysid(sid + '-500')
             except Exception as e:
                 yield Result(self, constants.ERROR,
-                             key=id,
+                             key='getnamebysid',
                              domain=domain,
                              error=str(e),
                              msg='Look up of ID {key} for {domain} failed: '
-- 
2.48.1

